src/diffusers/loaders/unet.py

```diff
@@ -597,16 +709,32 @@ def set_adapters(
 
         adapter_names = [adapter_names] if isinstance(adapter_names, str) else adapter_names
 
-        if weights is None:
-            weights = [1.0] * len(adapter_names)
-        elif isinstance(weights, float):
+        # Expand weights into a list, one entry per adapter
+        # examples for e.g. 2 adapters:  [7,7] -> [7,7] ; None -> [None, None]
+        if not isinstance(weights, list):
             weights = [weights] * len(adapter_names)
 
         if len(adapter_names) != len(weights):
             raise ValueError(
                 f"Length of adapter names {len(adapter_names)} is not equal to the length of their weights {len(weights)}."
             )
 
+        # Set None values to default of 1.0
+        # e.g. [7,7] -> [7,7] ; [None, None] -> [1.0, 1.0]
+        weights = [w if w is not None else 1.0 for w in weights]
+
+        blocks_with_transformer = {
+            "down": [i for i, block in enumerate(self.down_blocks) if hasattr(block, "attentions")],
+            "up": [i for i, block in enumerate(self.up_blocks) if hasattr(block, "attentions")],
+        }
+        transformer_per_block = {"down": self.config.layers_per_block, "up": self.config.layers_per_block + 1}
+
+        # e.g. [7,7] -> [{...}, {...}]
+        weights = [
+            self._expand_lora_scales_dict(weight_for_adapter, blocks_with_transformer, transformer_per_block)
+            for weight_for_adapter in weights
+        ]
+
```

Comment:
can we put the entire loop into a function and move it to a separate util file along with `_expand_lora_scales_dict`? this mean the logic of `blocks_with_transformers` and transformer_per_block` will also be moved there

```suggestion 
        weights = ._expand_lora_scales_dicts(weights)
```

Reply:
for
```python
weights = _expand_lora_scales_dicts(self, weights)
```
also let's make sure that if `weights` is a not dict, do nothing